src/transformers/models/tvp/configuration_tvp.py

```diff
@@ -0,0 +1,255 @@
+import copy
+import os
+from typing import Union
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+
+TVP_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "Intel/tvp-base": "https://huggingface.co/Intel/tvp-base/resolve/main/config.json",
+}
+
+
+class TVPVisionConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`TVPModel`]. It is used to instantiate an TVP
+    model according to the specified arguments, defining the model architecture. Instantiating a configuration with the
+    defaults will yield a similar configuration to that of the TVP
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+
+    Args:
+        input_format (`str`, *optional*, defaults to BGR):
+            The input format of image.
+        features (`List[str]`, *optional*, defaults to ["res5"]):
+            The feature extractor of the vision model. Other options can be "res2", "res3" and "res4".
+        resnets_depth (`int`, *optional*, defaults to 50):
+            The depth of resnets model which is the feature extractor.
+        resnets_num_groups (`int`, *optional*, defaults to 1):
+            The groups of torch.nn.Conv2d.
+        resnets_width_per_group (`int`, *optional*, defaults to 64):
+            The immediate conv layer channels per group.
+        resnets_stem_input_channels (`int`, *optional*, defaults to 3):
+            The input channels of stem module in resnet.
+        resnets_stem_out_channels (`int`, *optional*, defaults to 64):
+            The output channels of stem module and also the input channels of resnet stages.
+        resnets_res_out_channels (`int`, *optional*, defaults to 256):
+            The output channels of resnet.
+        resnets_res_dilation (`int`, *optional*, defaults to 1):
```

Comment:
The feature extractor / backbone should be loaded using the Backbone API using a backbone config e.g. [like this for DETR](https://github.com/huggingface/transformers/blob/fe861e578f50dc9c06de33cd361d2f625017e624/src/transformers/models/detr/modeling_detr.py#L350).